xmtpv3/crates/libxmtp-core/src/client.rs

```diff
@@ -1,13 +1,13 @@
 use crate::persistence::Persistence;
 
-pub struct Client<P: Persistence> {
-    persistence: P,
+pub struct Client {
+    persistence: Box<dyn Persistence>,
```

Comment:
For my own knowledge, whats the driver over the runtime determined type here rather the caller choosing it at compile time via generic?

Reply:
I think we should hold off on merging until we sanity-check the Box usage here. I have a PR (linked above) that just transfers the ownership of the underlying Persistence-implementing instance to the client. For this specific usage, I believe the ownership approach is simpler and avoids us having that `Send + Sync` race condition.

Thoughts @jazzz ?

Reply:
I had to write up some code to see if I'm crazy, and at least in the local_storage case we don't (unit test doesn't pass but example browser HTML test works): https://github.com/xmtp/libxmtp/compare/michaelx-local_storage_example?expand=1#diff-f3a6a7b2860bea48f5b4267aa7a8139b8bc48126dcb45424abd91d830cd41f90R26

For the iOS/Android cases, at least Mozilla Uniffi (one possible binding generator) does account for `Send + Sync` already in their callback interface: https://mozilla.github.io/uniffi-rs/udl/callback_interfaces.html#1-define-a-rust-trait because it can be useful. I can foresee us using a global Mutex to manage SQLite